c2dataviewer/control/scope_controller_base.py

```diff
@@ -313,6 +314,13 @@ def update_status(self):
         self.lastArrays = arraysReceived
         self.arrays = np.append(self.arrays, n)[-10:]
 
+        # Try to disconnect signal to prevent recursive calls of callback and then SegFaults.
```

Comment:
Thanks for finding that c2dataviewer wasn't safely updating the parameter tree. It looks like pyqtgraph already has a builtin way to prevent recursive calls to the tree change callback.  Can you try using [treeChangeBlocker](https://pyqtgraph.readthedocs.io/en/latest/api_reference/parametertree/parameter.html#pyqtgraph.parametertree.Parameter.treeChangeBlocker) in these cases instead of disconnecting and reconnecting the callback.  They also have blockTreeChangeSignal/unblockTreeChangeSignal methods, but using treeChangeBlocker is preferable